src/actions/sources/select.js

```diff
@@ -109,7 +114,17 @@ export function selectLocation(location: Location, tabIndex: string = "") {
       dispatch(closeActiveSearch());
     }
 
-    dispatch(addTab(source.toJS(), 0));
+    const src = source.toJS();
+    dispatch(
+      addTab(
+        {
+          id: src.id,
+          tooltip: getFileURL(src),
+          title: getFilename(src)
+        },
+        tabIndex
+      )
+    );
```

Comment:
do we need `addTab` here or could the tab reducer listen for `Select_SOURCE`? 

not sure :)